httpserver/service_node.cpp

```diff
@@ -1831,10 +1744,9 @@ std::string ServiceNode::get_status_line() const {
 }
 
 int ServiceNode::get_curr_pow_difficulty() const {
-
-    std::lock_guard guard(sn_mutex_);
-
-    return curr_pow_difficulty_.difficulty;
+    /// NOTE: difficulty is not longer used by modern clients, but
+    /// we send this to avoid breaking older clients.
+    return 1;
```

Comment:
The comment feels a bit misplaced: it should move to where this gets called when building the response in `request_handler.cpp`, which can just set it to `1` and then drop `get_curr_pow_difficulty()` from ServiceNode entirely.

Reply:
updated